client/post-editor/post-editor.jsx

```diff
@@ -156,6 +156,12 @@ export const PostEditor = React.createClass( {
 
 	componentWillReceiveProps: function( nextProps ) {
 		const { siteId, postId } = this.props;
+
+		// When switching posts, reset post edits for the post we're leaving
+		if ( nextProps.postId !== postId ) {
```

Comment:
A very edge case that might occur is switching from editing posts of the same ID on two different sites. Should we also check whether the site ID has changed?